src/services/services.ts

```diff
@@ -4316,6 +4361,159 @@ namespace ts {
                     }
                 }
             }
+
+            function getStringLiteralCompletionEntriesFromModuleNames(node: StringLiteral) {
+                const literalValue = node.text;
+                let result: CompletionEntry[];
+
+                const isRelativePath = startsWith(literalValue, ".");
+                const scriptDir = getDirectoryPath(node.getSourceFile().path);
+                if (isRelativePath || isRootedDiskPath(literalValue)) {
+                    result = getCompletionEntriesForDirectoryFragment(literalValue, scriptDir, getSupportedExtensions(program.getCompilerOptions()), /*includeExtensions*/false);
+                }
+                else {
+                    // Check for node modules
+                    result = getCompletionEntriesForNonRelativeModules(literalValue, scriptDir);
+                }
+
+                return {
+                    isMemberCompletion: false,
+                    isNewIdentifierLocation: false,
+                    entries: result
+                };
+            }
+
+            function getCompletionEntriesForDirectoryFragment(fragment: string, scriptPath: string, extensions: string[], includeExtensions: boolean): CompletionEntry[] {
+                // Complete the path by looking for source files and directories
+                const result: CompletionEntry[] = [];
+
+                const toComplete = getBaseFileName(fragment);

```

Comment:
In general, should we prefer completions with file extensions or without?

In other words, is the desired behavior something like this:
```typescript
// At first, completion list given entries with no extension
import d from ".\fo";         // Completes to .\foo

// As soon as we see a ".", switch to including the extension
import d from ".\foo.";       // Completes to .\foo.ts
```
If so, that's fine with me. That distinction should probably be made by the caller if they are the ones doing the filtering. This code should just always return results with extensions.